biz.aQute.bndlib/src/aQute/bnd/cdi/CDIAnnotations.java

```diff
@@ -1,55 +1,71 @@
 package aQute.bnd.cdi;
 
+import java.io.InputStream;
 import java.util.ArrayList;
 import java.util.Arrays;
-import java.util.Collection;
 import java.util.EnumSet;
 import java.util.List;
 import java.util.Map;
 import java.util.Map.Entry;
 import java.util.Set;
 import java.util.TreeSet;
 
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.xpath.XPath;
+import javax.xml.xpath.XPathConstants;
+import javax.xml.xpath.XPathExpression;
+import javax.xml.xpath.XPathExpressionException;
+import javax.xml.xpath.XPathFactory;
+
+import org.w3c.dom.Document;
+
 import aQute.bnd.component.MergedRequirement;
 import aQute.bnd.component.annotations.ReferenceCardinality;
 import aQute.bnd.header.Attrs;
 import aQute.bnd.header.OSGiHeader;
 import aQute.bnd.header.Parameters;
 import aQute.bnd.osgi.Analyzer;
 import aQute.bnd.osgi.Clazz;
+import aQute.bnd.osgi.Clazz.QUERY;
 import aQute.bnd.osgi.Constants;
 import aQute.bnd.osgi.Descriptors;
+import aQute.bnd.osgi.Descriptors.PackageRef;
 import aQute.bnd.osgi.Instruction;
 import aQute.bnd.osgi.Instructions;
+import aQute.bnd.osgi.Jar;
+import aQute.bnd.osgi.Packages;
+import aQute.bnd.osgi.Resource;
 import aQute.bnd.service.AnalyzerPlugin;
 import aQute.bnd.version.Version;
+import aQute.lib.exceptions.Exceptions;
 import aQute.lib.strings.Strings;
 
 /**
  * Analyze the class space for any classes that have an OSGi annotation for CCR.
  */
 public class CDIAnnotations implements AnalyzerPlugin {
 
-	public enum Discover {
-		all,
-		annotated,
-		annotated_by_bean,
-		none;
-
-		static void parse(String s, EnumSet<Discover> options, CDIAnnotations state) {
-			if (s == null)
-				return;
-			boolean negation = false;
-			if (s.startsWith("!")) {
-				negation = true;
-				s = s.substring(1);
-			}
-			Discover option = Discover.valueOf(s);
-			if (negation) {
-				options.remove(option);
-			} else {
-				options.add(option);
-			}
+	static final DocumentBuilderFactory	dbf	= DocumentBuilderFactory.newInstance();
```

Comment:
You can move `dbf` to be a local var in the static initialized since you don't need it any more after the static initializer completes. Best to let it be GC'd.